libaudit.go

```diff
@@ -315,12 +315,12 @@ done:
 			if err != nil {
 				return ret, err
 			}
-			if m.Header.Seq != seq {
-				// Wasn't the sequence number we are looking for, just discard it
+			if socketPID != os.Getpid() {
```

Comment:
This is wrong, as `os.Getpid()` won't match `socketPID` which is set to 0 while creating a netlink connection.
Similar kind of checks are deployed in auditd source code [see here](https://github.com/linux-audit/audit-userspace/blob/a12186cb8cb4632825feb695ee23dcca4e9d71a4/lib/netlink.c#L122) as well as golang netlink package [here](https://golang.org/src/syscall/netlink_linux.go#L91) . 

You'll see that auditd source code verifies that all received messages headers have pid set to 0.
We've also applied the same check although in a little different way.

Reply:
Yeah, I saw it. I think it's right to delete the comparison of socketPID.